### ISSUE TITLE
Crash when using collapse field as name

### ISSUE THREAD

**Issue by wmertens** [keystonejs/keystone#3225]
### Steps to reproduce the behavior

Make a list that has a collapsible name (I only tried with `map: {name: 'title'}` and the `title` collapses.

When you try to create an item, it crashes here: https://github.com/keystonejs/keystone/blob/master/fields/types/Field.js#L110

The reason is that it tries to lowercase the label, but with the name field the label is set to `false`.

Not sure how to fix this one, it's all a bit involved. Perhaps name fields should not be able to collapse.

**Comment by wmertens** [comment]
@jossmac @JedWatson thoughts?

**Comment by jossmac** [comment]
> Perhaps name fields should not be able to collapse.

👍

**Comment by JedWatson** [comment]
I agree, let's disable the `collapse` option on name fields. We should check for this in the `fields/Type` constructor so Keystone can log an explanation and refuse to start. Would you mind doing a PR for that @wmertens?

**Comment by JedWatson** [comment]
(and by "disable collapse on name fields" I mean disable `collapse` when a field has a `path` that's mapped to the `name` path on the `List`, this is nothing to do with `Types.Name` fields...)

**Comment by mxstbr** [comment]
> but with the name field the label is set to false.

This was, unrelated to this issue, possibly accidentally "fixed" in #3382?

![screen shot 2016-08-25 at 11 29 08](https://cloud.githubusercontent.com/assets/7525670/17953487/2b9e08a6-6ab7-11e6-80ba-c5fd3092459a.png)
